src/internal/resources/manager.cpp

```diff
@@ -32,22 +40,98 @@ Manager::Manager(std::unique_ptr<system::Resources> resources) :
   SystemProvider(*resources),
   m_system(std::move(resources))
 {
-    // for each host partition, construct the runnable resources
-    for (std::size_t i = 0; i < this->system().partitions().host_partitions().size(); ++i)
+    const auto& partitions      = system().partitions().flattened();
```

Comment:
Could you give a brief plain language overview of whats going on here?

What exactly is a host_partition in this context, and when we say construct xxx resource on a given partition, whats actually going on behind the scenes?

Reply:
Partitions are explained in more detail in "internal/system/partitions.hpp"

The system::System object owns 3 things:
- user options passed in
- topology info
- partition info

Partitions are formed by evaluating the user options against the topology information.

Initially we have two types of partitions for Host and Device. However, as mentioned in partitions.hpp, these are not a flat list, but have a hierarchy where in some cases, you can have multiple cuda devices attached to the same host partitions.

Later, when we define resources, like the "main" `FiberTaskQueue` or `LaunchControl` or the host `memory_resource`, those resources scale by host partition.

Other resources scale as the number of device partitions like the device `memory_resource`.

Finally resources like `ucx::Context` scale as number of device partitions + host partitions without devices.

This latter set is what we call the flattented partitions list or just  `system::Partitions`.  The are organized as a flat list sorted first by partitions that have a cuda device in cuda_device_id ordering from [0, N).  The remaining partitions in the "flattened" list correspond to any host regions without devices attached.